### PR TITLE
Add serverless plugin configuration validation 

### LINKED ISSUE #9: Use serverless plugins validation configuration feature
https://www.serverless.com/framework/docs/configuration-validation/

### PR DESCRIPTION
**Pull request by eoinsha**
Closes #9 

- Add schema generation
- Integration config validator into serverless plugin
- Add tests

### CODE REVIEW

File: serverless-plugin/index.js [hunk 1]
```diff
@@ -3,35 +3,26 @@
...
     this.options = options
     this.providerNaming = serverless.providers.aws.naming
 
-    const { topicArn, ...pluginConfig } = (
-      serverless.service.custom || {}
-    ).slicWatch
-    if (!topicArn) {
-      throw new Error('topicArn not specified in custom.slicWatch')
+    if (serverless.service.provider.name !== 'aws') {
+      throw new ServerlessError('SLIC Watch only supports AWS')
```
Comment (lmammino) [inline serverless-plugin/index.js:19]: good addition!

File: serverless-plugin/index.js [hunk 2]
```diff
@@ -42,6 +33,23 @@ class ServerlessPlugin {
    * Modify the CloudFormation template before the package is finalized
    */
   finalizeHook () {
+    const { topicArn, ...pluginConfig } = (
+      this.serverless.service.custom || {}
+    ).slicWatch
+    if (!topicArn) {
+      throw new ServerlessError('topicArn not specified in custom.slicWatch')
+    }
```
Comment (lmammino) [inline serverless-plugin/index.js:41]: Do we still need this or is it something we can make mandatory in the JSON schema validation?

File: serverless-plugin/render.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+'use strict'
```
Comment (lmammino) [inline serverless-plugin/render.js:1]: Does it make sense to commit this file?

File: serverless-plugin/config-schema.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+const schema = {
+  $schema: 'http://json-schema.org/draft-07/schema',
+  title: 'SLIC Watch Serverless Plugin configuration',
+  type: 'object',
+  properties: {
+    alarms: alarmsSchema,
+    dashboard: dashboardSchema,
+    topicArn: { type: ['string', 'null'] }
+  },
+  required: ['topicArn'],
```
Comment (lmammino) [inline serverless-plugin/config-schema.js:173]: If this one is required, do we need to support `null` as a type?

File: serverless-plugin/render.js (hunk 3 above)
Comment (eoinsha) [reply serverless-plugin/render.js:1]: No :)

File: serverless-plugin/config-schema.js (hunk 4 above)
Comment (eoinsha) [reply serverless-plugin/config-schema.js:173]: Nice catch, it should not support null!

File: serverless-plugin/index.js [hunk 5]
```diff
@@ -1,37 +1,31 @@
...
+const Ajv = require('ajv')
+
 const alarms = require('./alarms')
 const dashboard = require('./dashboard')
+const { pluginConfigSchema, slicWatchSchema } = require('./config-schema')
 const defaultConfig = require('./default-config')
 const CloudFormationTemplate = require('./cf-template')
 
+const ServerlessError = require('serverless/lib/serverless-error')
+const normalizeAjvErrors = require('serverless/lib/classes/ConfigSchemaHandler/normalizeAjvErrors')
```
Comment (lmammino) [inline serverless-plugin/index.js:13]: I am wondering how _safe_ (i.e. backward-compatible) this is...

What happens if Serverless framework shuffles things around internally in a new version and these modules are not there anymore?

We currently have `2.x` of `serverless` as a peer dependency... Should we move it to an actual regular dependency?

BTW, this is minor for now. Maybe just food for thought. We could probably hit the bullet and go ahead with this. If there's a breaking change we can deal with it later and find an appropriate solution when we have more context...

**Review by lmammino — APPROVED** [review]
Great stuff! 👍🏽

File: serverless-plugin/index.js (hunk 5 above)
Comment (eoinsha) [reply serverless-plugin/index.js]: Good point! TIL Ajv has an `errorsText` function so I've replaced it with this!

**Pull request merged**